File: parley/src/main/java/nu/parley/android/data/net/service/DeviceService.java
package nu.parley.android.data.net.service;

import nu.parley.android.data.model.Device;
import nu.parley.android.data.net.RepositoryCallback;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface DeviceService {

    void register(RepositoryCallback<Void> callback, Device device);

    @POST("devices")
    Call<Void> register(@Body Device device);

}

File: parley/src/main/java/nu/parley/android/data/net/service/DeviceService.kt
package nu.parley.android.data.net.service

import nu.parley.android.data.model.Device
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

internal interface DeviceService {

    @POST("devices")
    fun register(@Body device: Device): Call<Void>
}
